dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -204,31 +210,79 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
+    // Retrieve DisplayName and IconUri
+    // 1. We call into Shell APIs, to get both from the app taskbar shortcut.
+    // 2. If 1 doesn't work (i.e. possibly because the developer didn't specify any DisplayName/Icon,
+    //    we take the file name as app name, and let Shell to put a default icon
+    winrt::com_ptr<IPropertyStore> propertyStore;
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
+    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
+
+    // Retrieve the display name
+    std::wstring displayName{};
+
+    wil::unique_prop_variant propVariantDisplayName;
+    THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
```

Comment:
Don't throw if this isn't found. Instead default to the ProcessName.

Reply:
The only manadatory property is IconUri.

Reply:
Ah, that's true. Good catch!